packages/react-bindings/src/styles/resolveStylesAndClasses.tsx

```diff
@@ -3,71 +3,158 @@ import {
   isDebugEnabled,
   ICSSInJSStyle,
   ComponentStyleFunctionParam,
+  ComponentSlotStylesResolved,
+  ThemePrepared,
 } from '@fluentui/styles'
 import { ComponentSlotClasses } from '../styles/types'
 
-// Both resolvedStyles and classes are objects of getters with lazy evaluation
+export type ResolveStylesResult = {
+  resolvedStyles: ComponentSlotStylesResolved
+  resolvedStylesDebug: Record<string, { styles: Object }[]>
+  classes: ComponentSlotClasses
+}
+
+// TODO: Describe me
+
+const classesCache = new WeakMap<ThemePrepared, Record<string, string>>()
+const stylesCache = new WeakMap<ThemePrepared, Record<string, ICSSInJSStyle>>()
+
+/**
+ * Both resolvedStyles and classes are objects of getters with lazy evaluation
+ *
+ * Additionally if the cacheEnabled option is provided, than the resolved styles
+ * and classes are caching the results in WeakMaps. The key of the maps contains the following:
+ * - theme
+ * - displayName
+ * - slot name
+ * - styling props
+ * - rtl mode
+ * - disable animations mode
+ */
 const resolveStylesAndClasses = (
   mergedStyles: ComponentSlotStylesPrepared,
   styleParam: ComponentStyleFunctionParam,
   renderStyles: (styles: ICSSInJSStyle) => string,
-): {
-  resolvedStyles: ICSSInJSStyle
-  resolvedStylesDebug: Record<string, { styles: Object }[]>
-  classes: ComponentSlotClasses
-} => {
+  cacheEnabled?: boolean | undefined,
+  displayName?: string,
+  theme?: ThemePrepared,
+  props?: object,
```

Comment:
Can we somehow reduce the amount of passed things?

Reply:
We can extract these to config object, but it seems awkward to have part of the properties first line arguments, and than part of them in config object.. The other options is to move everything to config object - maybe this one is better, what do you think?

Reply:
Can we merge `resolveStyles` and `resolveStylesAndClasses` functions as I don't understand the separation there anymore. I think that I will solve params issue.